### DISCUSSION TITLE
Safe Isaac Gym support

### DISCUSSION THREAD

**Discussion by iveta331** [PKU-Alignment/safety-gymnasium#83] [Q&A]
Hello guys, 
first of all, this is a great repo, I'm so glad that i don't have to struggle with making safety gym work anymore 😄 
However, I wanted to ask, are environments from Safe Isaac Gym already officially supported?
When i try to initialize an env (for example env = safety_gymnasium.make('FreightFrankaCloseDrawer')) I keep getting the "AssertionError: Environment FreightFrankaCloseDrawer is not registered in safety-gymnasium.". Same for all Safe Isaac Gym envs.
However, looking at the docs, it seems that there is a lot of info missing, for other envs, there is the "Import" info for how to initialize envs, not here though. Makes me wonder that maybe I am initializing it wrong? But the FreightFrankaCloseDrawer is listed in the table here on GitHub as an example to Safe Isaac Gym env, so I thought that it should work.
Also, looking at the scripts in Examples folder, there is no example for this. So is it not supported yet, or is it a bug, or am I doing something wrong?

**Comment by muchvo** [comment]
Thank you for your recognition and feedback. We will be updating some related content about Isaac Gym installation and environment setup soon. To answer your question, the backend of Safe Isaac Gym is different from other environments. Therefore, you need to first [install Isaac Gym](https://developer.nvidia.com/isaac-gym). Additionally, because Isaac Gym's mechanics significantly differ from MuJoCo, the way to invoke the Isaac Gym environment library usually follows Nvidia's example style, which is also the case in our environment. Regarding running the environment, you can refer to [this code](https://github.com/PKU-Alignment/Safe-Policy-Optimization/blob/main/safepo/common/env.py). Furthermore, SafePO supports the direct execution of Safe Isaac Gym, providing you with a quick training example and script. Any further questions are also welcome.

**Reply by iveta331** [reply]
Thank you, I think I already understand how it's supposed to work. 🙂

**Comment by zhaoxuesi** [comment]
I have the same problem when I use the "Safety[Agent]MultiGoal1-v0" in Multigoal. I get the error as "AssertionError: Environment SafetyAntMultiGoal1-v0 is not registered in safety-gymnasium." I would be most grateful if someone could help me.
![picture of document](https://github.com/user-attachments/assets/98aba555-7555-4724-b0c5-2298e130452f)
